### PR TITLE
Rewire batch submission.  Closes #1017

### LINKED ISSUE #1017: Make Batch Submit API great again

### PR DESCRIPTION
**Pull request by mcovarr**
The README and Swagger were already great, so no changes there.  The /batch name of the endpoint is maybe not so great; is this perhaps how this was overlooked?  The code was so close to greatness already.

### CODE REVIEW

**Comment by cjllanwarne** [comment]
Somebody Else's Code: I would prefer an immutable var instead of a mutable val

**Comment by geoffjentry** [comment]
@cjllanwarne There are tradeoffs on that conunudrum. Can you explain your preference?

File: engine/src/main/scala/cromwell/webservice/WorkflowSubmitBatchActor.scala [hunk 1]
```diff
@@ -34,7 +35,9 @@ case class WorkflowManagerBatchIndexedResponse(response: WorkflowManagerResponse
 class WorkflowSubmitIndexedActor(batchSubmitActor: ActorRef, requestHandlerActor: ActorRef,
```
Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/webservice/WorkflowSubmitBatchActor.scala]: At the risk of jumping onto the Thibault bandwagon... `Router`?

OTOH... nah :)

Comment (mcovarr) [inline engine/src/main/scala/cromwell/webservice/WorkflowSubmitBatchActor.scala]: IMHO that's a fine suggestion for an enhancement ticket, but beyond the scope of this "make it great again" PR.  😄

**Comment by cjllanwarne** [comment]
@geoffjentry I want to resort to authority and say "Zen of Akka"... 

Reasons for my gut feeling: A mutable val makes it look and act more like a state machine, and reduces the risk of accidentally leaking the variable pointer to other threads which may update it out of band. Obviously not likely in this case, but as a muscle memory thing a-la `Some(constant)`

**Comment by cjllanwarne** [comment]
👍

**Comment by Horneth** [comment]
👍

**Pull request merged**